desc/objectives/getters.py

```diff
@@ -102,58 +105,24 @@ def get_fixed_axis_constraints(
         FixPsi(eq=eq, normalize=normalize, normalize_target=normalize),
     )
     if profiles:
-        if kinetic:
-            constraints += (
-                FixElectronDensity(
-                    eq=eq, normalize=normalize, normalize_target=normalize
-                ),
-                FixElectronTemperature(
-                    eq=eq, normalize=normalize, normalize_target=normalize
-                ),
-                FixIonTemperature(
-                    eq=eq, normalize=normalize, normalize_target=normalize
-                ),
-                FixAtomicNumber(eq=eq, normalize=normalize, normalize_target=normalize),
-            )
-        else:
-            constraints += (
-                FixPressure(eq=eq, normalize=normalize, normalize_target=normalize),
-            )
-            if anisotropy:
+        for name, con in _PROFILE_CONSTRAINTS.items():
+            if getattr(eq, name) is not None:
```

Comment:
So if the attribute is not in the dictionary, would it be free?

Reply:
the dictionary covers all profiles types, so it should fix anything that the equilibrium has

Reply:
Not free, if an equilibrium does not have that profile assigned, then it does not have that attribute at all

i.e. the `FixIota` constraint will not be given to an equilibrium which does not have an `eq.iota` attribute, but if it lacks that attribute that means that instead it has a `eq.current` attribute instead and that one will be fixed

same for kinetic profiles(n and T) and pressure, if the former are present then the latter must not be present, and vice-versa